docs/wfforward.md

```diff
@@ -0,0 +1,1075 @@
+
+* Define genomic regions as integers of intervals, $R \in {0, 1, 2, \ldots, L-1}$, where $L$ is the region length.
+* Use functions designed to sample from ranges of integers to generate mutation positions.  For example, [`np.random.randint`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.random.randint.html).
+
+### Summary
+* Our simplification function is getting more complex and taking more arguments.  At this point, we'd help ourselves by encapsulating some of this work into a proper class, which we do below.
+* `collections.namedtuple` plus pickling is a straightforward method to handle metadata in Python
+
+## Starting with a prior history
+
+In this section we show how to 'seed' a forward simulation with a set of nodes and edges generated from a coalescent simulation.
+
+Let's take a look at how many roots are present in each marginal tree that came out of our simulation with recombination in the previous section:
+
+
+```python
+ts = msprime.load_tables(nodes=n,edges=e,sites=s,mutations=m)
+
+nroots=[]
+for i in ts.trees():
+    nroots.append(i.num_roots)
+p = sns.distplot(nroots, kde=False)
+p.set(xlabel="Number of roots",ylabel="Number");
+```
+
+
+![svg](wfforward_files/wfforward_51_0.svg)
+
+
+Most of our trees have one root, but a handful have two.  What is happening is that even though we simulated for $10N$ generations, and $E[TMRCA]=4N$, the variance is pretty big, such that some marginal trees are not completely coalesced.  Let's take a look at the distribution of TMRCA using msprime:
+
+
+```python
+tmrca=[]
+sims=msprime.simulate(100,recombination_rate=25.0,random_seed=42,num_replicates=10000)
+for sim in sims:
+    for marginal in sim.trees():
+        tmrca.append(marginal.get_time(marginal.get_root()))
+p = sns.distplot(tmrca, kde=False)
+p.set(xlabel="TMRCA (units of N generations)",ylabel="Number");
+```
+
+
+![svg](wfforward_files/wfforward_53_0.svg)
+
+
+So, some fraction of marginal trees have *very* large TMRCA! 
+
+When running forward simulations and *not* simulating neutral mutations, uncoalesced marginal trees matter.  These "sub-trees" (the disjoint ancestries within a non-recombining part of the population history) have been simplified such that the most ancient node on a sub-tree is the most-recent time poing required to represent that lineage.  In other words, the sub trees *do not* go all the way back to one of the original $2N$ nodes used to start the simulation.  The implication is that you will get an incorrect total number of neutral mutations when you add mutations to your simulation at the end with an `msprime.MutationGenerator` instance.
+
+We have threeways to deal with this:
+
+* Run the forward simulations longer. (Boo! Hiss!)
+* Start the simulation with a (reasonable) coalescent tree.
+* Retain the original $2N$ nodes as samples.
```

Comment:
This is clever, but I think the branch length distribution would still be wrong...

Reply:
Good point.  I'll make a note of this and attempt to test it.  I think you'll get too many mutations.  Hmmm--this is a gotcha for folks wanting to do wacky models...